pkg/controller/certificaterequests/venafi/venafi.go

```diff
@@ -88,7 +89,20 @@ func (v *Venafi) Sign(ctx context.Context, cr *cmapi.CertificateRequest, issuerO
 
 	duration := apiutil.DefaultCertDuration(cr.Spec.Duration)
 
-	certPem, err := client.Sign(cr.Spec.CSRPEM, duration)
+	var customFields []venafiinternal.CustomField
+	if annotation, exists := cr.GetAnnotations()[cmapi.VenafiCustomFields]; exists && annotation != "" {
+		err := json.Unmarshal([]byte(annotation), &customFields)
+		if err != nil {
+			message := "Failed to parse venafi.cert-manager.io/custom-fields annotation"
+
+			v.reporter.Failed(cr, err, "CustomFieldsError", message)
+			log.Error(err, message)
+
+			return nil, err
```

Comment:
We shouldn't return the error here as we have 'handled' it by marking the CertificateRequest as failed - by returning an error, we immediately trigger a 're-queue' of the resource in the controller's workqueue which is unnecessary as calling `reporter.Failed` will cause an update to the resource so it will be automatically re-synced anyway 😄

Reply:
Interesting! Didn't know